packages/components/Button/src/Button.tsx

```diff
@@ -34,19 +34,56 @@ export const buttonLookup = (layer: string, state: IPressableState, userProps: B
   );
 };
 
+export const extractOuterStylePropsAndroid = memoize((style) => {
+  const marginKeys = [
+    'margin',
+    'marginTop',
+    'marginBottom',
+    'marginLeft',
+    'marginRight',
+    'marginStart',
+    'marginEnd',
+    'marginVertical',
+    'marginHorizontal',
+    'start',
+    'end',
+    'left',
+    'right',
+    'top',
+    'bottom',
+  ];
+  const outerStyleProps = {},
+    resetProps = {};
+  marginKeys.forEach((key) => {
+    if (style && style[key]) {
+      outerStyleProps[key] = style[key];
+      resetProps[key] = 0;
```

Comment:
I think "undefined" is a better "reset".

Also it seems like it'd be cleaner to do something like

```
const {<all outer props>, ...restStyles} = style;

return [
   {<all outer props>},
   restStyles
]
```

Reply:
I was wondering if it would be fine to repeat the keys twice but turns out it's actually much much cleaner, going for this!